test/canary/scripts/run_test.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# cleanup on EXIT regardles of error 
+
+# Inputs to this file as environment variables
+# SERVICE
+# SERVICE_REGION
+# CLUSTER_REGION
+# CLUSTER_NAME
+# SERVICE_REPO_PATH
+# NAMESPACE
+
+set -euo pipefail
+export NAMESPACE=${NAMESPACE:-"ack-system"}
+export AWS_DEFAULT_REGION=$SERVICE_REGION 
+export E2E_DIR=$SERVICE_REPO_PATH/test/e2e/
+SCRIPTS_DIR=${SERVICE_REPO_PATH}/test/canary/scripts
+
+source $SCRIPTS_DIR/setup_oidc.sh
+source $SCRIPTS_DIR/install_controller_helm.sh
+
+function print_controller_logs() {
+  pod_id=$( kubectl get pods -n $NAMESPACE --field-selector="status.phase=Running" \
+      --sort-by=.metadata.creationTimestamp \
+      | grep ack-applicationautoscaling-controller | awk '{print $1}' 2>/dev/null )
+
+  kubectl -n $NAMESPACE logs "$pod_id"
+}
+
+function cleanup {
+  echo "Cleaning up resources"
+  set +e
+  kubectl delete scalingpolicies --all
+  kubectl delete scalabletargets --all
+
+  print_controller_logs
+
+  helm delete -n $NAMESPACE ack-$SERVICE-controller
+  pushd $SERVICE_REPO_PATH
+    kubectl delete -f helm/crds
+  popd
+  kubectl delete namespace $NAMESPACE
+
+  cd $E2E_DIR
+  export PYTHONPATH=.. 
+  python service_cleanup.py
+
+}
+trap cleanup EXIT
+
+# Update kubeconfig
+aws --region $CLUSTER_REGION eks update-kubeconfig --name $CLUSTER_NAME
+
+# Setup OIDC
+create_oidc_role "$CLUSTER_NAME" "$CLUSTER_REGION" "$NAMESPACE"
+
+# Install service helm chart
+install_helm_chart $SERVICE $OIDC_ROLE_ARN $SERVICE_REGION $NAMESPACE
+
+echo "Log helm charts are deployed properly"
+kubectl -n $NAMESPACE get pods
+kubectl get crds
+
+pushd $E2E_DIR
+  export PYTHONPATH=..
+  # create resources for test
+  python service_bootstrap.py
+  sleep 5m
```

Comment:
Why?

Reply:
at the time of writing this script, the jobs sometimes weren't able to find s3 files due to no-blocking s3 file upload 
that's why sleep was necessary. 
Not sure if this issue is fixed now after you moved test code

Reply:
I remember asking or at-least wondering the same thing, should not be needed but maybe @surajkota saw some issues, I will let him answer this one.

Reply:
yes, still need it